src/build_context.rs

```diff
@@ -709,14 +712,16 @@ impl BuildContext {
             }
         }
 
-        // I wouldn't know of any case where this would be the wrong (and neither do
-        // I know a better alternative)
-        let bin_name = artifact
-            .file_name()
-            .expect("Couldn't get the filename from the binary produced by cargo");
-        self.add_external_libs(&mut writer, artifact, ext_libs)?;
+        for (artifact, artifact_ext_libs) in artifacts.iter().zip(ext_libs) {
+            // I wouldn't know of any case where this would be the wrong (and neither do
+            // I know a better alternative)
+            let bin_name = artifact
+                .file_name()
+                .expect("Couldn't get the filename from the binary produced by cargo");
+            self.add_external_libs(&mut writer, artifact, artifact_ext_libs)?;
```

Comment:
TODO: Multiple binaries may have same external shared library dependencies, so we should only bundle one version.